homeassistant/components/pyload/config_flow.py

```diff
@@ -0,0 +1,122 @@
+"""Config flow for pyLoad integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from aiohttp import CookieJar
+from pyloadapi.api import PyLoadAPI
+from pyloadapi.exceptions import CannotConnect, InvalidAuth, ParserError
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_SSL,
+    CONF_USERNAME,
+    CONF_VERIFY_SSL,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_create_clientsession
+import homeassistant.helpers.config_validation as cv
+
+from .const import DEFAULT_HOST, DEFAULT_NAME, DEFAULT_PORT, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Required(CONF_PORT, default=DEFAULT_PORT): cv.port,
+        vol.Required(CONF_SSL, default=False): cv.boolean,
+        vol.Required(CONF_VERIFY_SSL, default=True): bool,
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, user_input: dict[str, Any]) -> None:
+    """Validate the user input and try to connect to PyLoad."""
+
+    session = async_create_clientsession(
+        hass,
+        user_input[CONF_VERIFY_SSL],
+        cookie_jar=CookieJar(unsafe=True),
+    )
+
+    url = (
+        f"{"https" if user_input[CONF_SSL] else "http"}://"
+        f"{user_input[CONF_HOST]}:{user_input[CONF_PORT]}/"
+    )
+    pyload = PyLoadAPI(
+        session,
+        api_url=url,
+        username=user_input[CONF_USERNAME],
+        password=user_input[CONF_PASSWORD],
+    )
+
+    await pyload.login()
+
+
+class PyLoadConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for pyLoad."""
+
+    VERSION = 1
+    # store values from yaml import so we can use them as
+    # suggested values when the configuration step is resumed
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            self._async_abort_entries_match(
+                {CONF_HOST: user_input[CONF_HOST], CONF_PORT: user_input[CONF_PORT]}
+            )
+            try:
+                await validate_input(self.hass, user_input)
+            except (CannotConnect, ParserError):
+                errors["base"] = "cannot_connect"
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+            except Exception:
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
+                title = user_input.pop(CONF_NAME, None)
```

Comment:
The second argument of pop is "if it does not exist" right? Should we add the default name there?